src/ray/gcs/gcs_worker_manager.h

```diff
@@ -81,6 +90,16 @@ class GcsWorkerManager : public rpc::WorkerInfoGcsServiceHandler {
   std::vector<std::function<void(std::shared_ptr<rpc::WorkerTableData>)>>
       worker_dead_listeners_;
 
+  /// A deque where workers are store as pairs of (WorkerID, Timestamp).
+  /// The workers are sorted according to the timestamp, and the oldest is at the head of the deque.
+  /// @note The pair consists of:
+  ///   - first: WorkerID (identifier for the worker)
+  ///   - second: Timestamp (time when the worker was last updated or added)
+  std::deque<std::pair<WorkerID, int64_t>> sorted_dead_worker_deque_;
```

Comment:
Speaking generally to the entire PR's approach, the fix I'd MUCH rather see is that we augment the behavior of the table storage in GCS itself.  The problem that is being examined here isn't specific to worker information, it's actually an issue for nearly all the tables in GCS.  So instead of augmenting this single class, let's see if we can have a useful new feature in table storage itself.

Reply:
Essentially, I'd want to leverage an EXPIRE's or TTL policy or size based policy for the in memory storage, and then for the situation where we're using Redis, leverage Redis features like setting EXPIRES for a key to do clean up.